backend/git_real/tests/test_api_views.py

```diff
@@ -1,13 +1,68 @@
 from rest_framework.test import APITestCase
 from test_mixins import APITestCaseMixin
+from .factories import RepositoryFactory
+from .utils import get_body_and_headers
+from django.urls import reverse
+from django.utils import timezone
+import dateutil.parser
+import mock
 
-
-from . import factories
+from git_real import views
+from git_real.models import Push
+from git_real.permissions import IsWebhookSignatureOk
+from git_real.helpers import github_timestamp_int_to_tz_aware_datetime
 
 
 class RepositoryViewsetTests(APITestCase, APITestCaseMixin):
     LIST_URL_NAME = "repository-list"
     SUPPRESS_TEST_POST_TO_CREATE = True
 
     def verbose_instance_factory(self):
-        return factories.RepositoryFactory(archived=True)
+        return RepositoryFactory(archived=True)
+
+
+class PushEventTests(APITestCase):
```

Comment:
It's important to also check that the view actually exposes data. The api endpoint isn't just concerned with permissions, it neds to return actual data.

Please make use of the api mixins to test that. Just use SUPPRESS_POST_TO_CREATE